File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution(object):
    def maxArea(self, height):
        l,r=0,len(height)-1
        res=0
        while l<r:
            area=(r-l)*min(height[l],height[r])
            res=max(res,area)

            if height[l]<height[r]:
                l+=1
            else:
                r-=1

        return res

File: Data Structures & Algorithms/max-water-container/submission-0.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int left=0;
        int right=heights.size()-1;
        int water=0;
        while(left<right){
            int height_left=heights[left];
            int height_right=heights[right];
            int area = min(height_left, height_right) * (right - left);
            if (area > water) {
                water = area;
            }
            if(height_left<height_right){
                left++;
            }else{
                right--;
            }
            }
                return water;
        }
};
